src/ews_serialize.erl

```diff
@@ -339,9 +442,30 @@ match_children_elems([], [#elem{meta=#meta{min=0}}=E|Es], [], Res) ->
     match_children_elems([], Es, [], [{undefined,E}|Res]);
 match_children_elems([], [#elem{meta=#meta{min=0}}=E|Es], Acc, Res) ->
     match_children_elems([], Es, [], [{undefined,E},lists:reverse(Acc)|Res]);
+match_children_elems([], [#elem{qname=Name,meta=#meta{min=_N}}|_], _, _) ->
+    error({missing_non_optional_element, Name});
 match_children_elems([], [], Acc, Res) ->
     [lists:reverse(Acc) | lists:reverse(Res)].
 
+validate_attrs([{?SCHEMA_INSTANCE_NS, _} | As], PossAttrs, Acc) ->
+    validate_attrs(As, PossAttrs, Acc);
+validate_attrs([{Name, Value} | As], PossAttrs, Acc) ->
+    case [ P || #attribute{name = {_, N}} = P <- PossAttrs, N == Name ] of
+        [] ->
+            logger:notice("Unexpected attribute: ~p~n", [Name]),
+            validate_attrs(As, PossAttrs, Acc);
+        [#attribute{}] ->
+            %% TODO: validate the type
+            %% TODO: how to we handle utf8 in both Name and Value?
```

Comment:
Since `Name` and `Value` are just lists, does this mean they are lists of Unicode code points or Latin 1 characters?

With this solution proposed in this PR, what happens if there is a utf8 encoded value for an attribute?

Reply:
It should work it turns out.

```17> list_to_atom("föö").      
föö
18> binary_to_atom(<<"föö"/utf8>>).
föö
```

Reply:
So just remove the comment?